src/main/java/uk/gov/ida/stub/idp/repositories/SessionRepositoryBase.java

```diff
@@ -0,0 +1,133 @@
+package uk.gov.ida.stub.idp.repositories;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.joda.JodaModule;
+import org.jdbi.v3.core.Jdbi;
+import org.opensaml.core.xml.XMLObject;
+import org.opensaml.saml.saml2.core.AuthnContextComparisonTypeEnumeration;
+import uk.gov.ida.common.SessionId;
+import uk.gov.ida.saml.hub.domain.IdaAuthnRequestFromHub;
+import uk.gov.ida.stub.idp.exceptions.SessionSerializationException;
+import uk.gov.ida.stub.idp.repositories.jdbc.mixins.AuthnContextComparisonTypeMixin;
+import uk.gov.ida.stub.idp.repositories.jdbc.mixins.IdaAuthnRequestFromHubMixin;
+import uk.gov.ida.stub.idp.repositories.jdbc.mixins.XmlObjectMixin;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Optional;
+
+import static com.fasterxml.jackson.databind.DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES;
+import static org.apache.commons.lang3.StringEscapeUtils.unescapeJson;
+
+public abstract class SessionRepositoryBase<T extends Session> implements SessionRepository<T> {
+    private final Jdbi jdbi;
+    private final ObjectMapper objectMapper;
+    private final Class<T> sessionType;
+
+    public SessionRepositoryBase(Class<T> sessionType, Jdbi jdbi) {
+        this.sessionType = sessionType;
+        this.jdbi = jdbi;
+        this.objectMapper = createObjectMapperForSerialization();
+    }
+    
+    public boolean containsSession(SessionId sessionToken) {
+        return jdbi.withHandle(handle -> handle.select(
+                "select count(1) from stub_idp_session where session_id = ?", sessionToken.toString())
+                .mapTo(Boolean.class)
+                .findOnly());
+    }
+    
+    public Optional<T> get(SessionId sessionToken) {
+        Optional<String> sessionData = jdbi.withHandle(handle -> handle.select(
+                "select session_data from stub_idp_session where session_id = ?", sessionToken.toString())
+                .mapTo(String.class)
+                .findFirst());
+
+        if (!sessionData.isPresent()) {
```

Comment:
There is an ifPresent method on Optional that will execute the method/lambda in ifPresent or return Optional.empty

Reply:
We use the `isPresent` method to return an empty Optional early if there's no session data. As there isn't a `isNotPresent` method on Optional and that you can't negate `isPresent`, I think it's probably best to keep this call as it is.